encryption.tex

```diff
@@ -0,0 +1,202 @@
+Since the drive must me encrypted with LUKS early during the provisioning, new Kickstart Provisioning Template and Partition Tables had to be created at Foreman.
+
+\vskip 0.5cm
+\begin{lstlisting}[language=bash]
+#Encrypted VDA - Partition Table
+
+ignoredisk --only-use=${BOOT_DEV}
+zerombr
+clearpart --drives=${BOOT_DEV} --all --initlabel
+part /boot     --size=1024 --asprimary --ondrive=${BOOT_DEV}
+part /boot/efi --size=200  --asprimary --ondrive=${BOOT_DEV} --fstype=efi
+part pv.boot   --size=1 --grow  --encrypted --passphrase=temppass --ondisk=${BOOT_DEV}
+volgroup ${BOOT_VG} pv.boot
+logvol /               --vgname=${BOOT_VG} --size=1 --grow --name=root
+\end{lstlisting}
+
+"Encrypted VDA" initialize the System disk with two regular partitions - /boot and /boot/efi - and then a PV, a VG and a LV, been the LV encrypted through LUKS with a temporary password
+\begin{lstlisting}[language=bash]
+##Kickstart - Encrypted Provisioning Template
+#Packages Section
+%packages
+clevis-dracut
+#Post Section
+%post --log=/mnt/sysimage/root/install.post.log
+curl -sfg http://tang01.cp.lsst.org/adv -o adv1.jws
+clevis luks bind -f -k- -d /dev/vda3 \
+tang '{"url":"http://tang01.cp.lsst.org","adv":"adv1.jws"}' <<< "temppass"
+curl -sfg http://tang02.cp.lsst.org/adv -o adv2.jws
+clevis luks bind -f -k- -d /dev/vda3 \
+tang '{"url":"http://tang02.cp.lsst.org","adv":"adv2.jws"}' \ <<< "temppass"
+cryptsetup luksRemoveKey /dev/vda3 <<< "temppass"
+\end{lstlisting}
+
+In the packages section, clevis-dracut is installed, to then be used at post to communicate with a Tang server(s), subscribe to them and remove the temporary password.
+
+\newpage
+\subsection{Test Environment}
+\begin{itemize}
+  \item Two Tang servers using the tang puppet profile.
+  \item A client with the clevis puppet profile.
+  \item The client VM (clevis01.cp.lsst.org) is provisioned through PXE with 'Encrypted VDA' Partitioning Table and 'Kickstart Encrypted Provisioning Template'.
+  \item During partition creation, clevis01 root partition is encrypted through LUKS with the passphrase 'temppass'.
+  \item Then at packages, clevis-dracut is installed to then communicate with the Tang servers at post section.
+  \item At post, clevis01 subscribes to the Tang servers (tang01.cp.lsst.org and tang02.cp.lsst.org) and the temporary passphrase encryption key is removed as a decryption mechanism.
+\end{itemize}
+
+\begin{figure}
+  \includegraphics[width=10cm]{images/image4.png}
+  \centering
+  \caption{Booting procedure for an enrolled or newly enrolled client.}
+\end{figure}
+
+\begin{enumerate}
+  \item During boot, the client machine attempts to reach first Tang server.
+  \item If reached, the decryption server hands over the decryption key.
+  \item If the first Tang server wasn't reachable, it attempts with the next one in the key slot.
+  \item The second Tang server sends the decryption key.
+\end{enumerate}
+
+\newpage
+\subsection{Lab Results}
```

Comment:
I'm guessing that the tang server VMs either had no disk encryption or required a manual passphrase to be entered into the console on startup?  I'm fairly sure that deploying a system that requires manual human intervention in order to "reboot" on its own is not a good idea.  I would like to see tang server integration with a TPM proven before we make a final decision.

Reply:
Correct, neither of the tang servers has the disk encrypted. Not following how you expect the Trusted Policy to be applied to the tang server.